pandas/tests/frame/test_operators.py

```diff
@@ -891,6 +891,22 @@ def test_no_warning(self, all_arithmetic_operators):
 
 
 class TestTranspose:
+    @pytest.mark.parametrize(
+        "ser",
+        [
+            pd.date_range("2016-04-05 04:30", periods=3, tz="UTC"),
+            pd.period_range("1994", freq="A", periods=3),
+            pd.period_range("1969", freq="9s", periods=1),
+            pd.date_range("2016-04-05 04:30", periods=3).astype("category"),
+            pd.date_range("2016-04-05 04:30", periods=3, tz="UTC").astype("category"),
+        ],
+    )
+    def test_transpose_retains_extension_dtype(self, ser):
+        # case with more than 1 column, must have same dtype
+        df = pd.DataFrame({"a": ser, "b": ser})
+        result = df.T
+        assert (result.dtypes == ser.dtype).all()
+
```

Comment:
```
round_trip = df.T.T
tm.assert_frame_equal(df, round_trip)
```
?

Reply:
Interestingly, the `pd.date_range("2016-04-05 04:30", periods=3).astype("category")` case fails that test. All the values are `NaT`.

I've xfalied it for now, and likely won't have time to look into it.

Reply:
sounds good, thanks